tests/integration/test_model.py

```diff
@@ -1032,3 +1054,130 @@ def test_predict_with_nullable_types(c):
         result,
         check_dtype=False,
     )
+
+
+# TODO - many ML tests fail on clusters without sklearn - can we avoid this?
+@xfail_if_external_scheduler
```

Comment:
Hi @charlesbluca I'm not sure why this test is failing on the Dask cluster job when `@xfail_if_external_scheduler` is here (it worked before with `@skip_if_external_scheduler`)?

Reply:
looks like the test was xpassing. Can we try removing the xfail if and seeing if things changed?

Reply:
Ohh I see, thanks!

Reply:
Looks like these tests are passing now. @charlesbluca Idk if we started including sklearn in the container or if some of the changes to re-use the fixture at a file level changed this. But things seem to work now

Reply:
This test is `test_ml_class_mappings`, right? If so I would expect things to pass here on an independent cluster since it seems like all the test is doing is importing the given classes on the client-side, which should have all the relevant ML libraries installed.

Was there a run when this test was failing in the cluster testing but passing elsewhere?

Reply:
I think originally I had all of the tests in a single method, and then when I broke them out into `test_ml_class_mappings`, `test_agnostic_cpu_xgb_models`, and `test_agnostic_gpu_xgb_models` I just kept the skip/xfail decorator on all of them. It took switching to xfail to catch the xpass 🙂